### DISCUSSION TITLE
How to add javascript code in <script> in <head> ?

### DISCUSSION THREAD

**Discussion by wzdxy** [nuxt/nuxt#17994] [Questions]
I need exec some js script on the head (calc rem by dpr and screenWidth, for mobile adaptation)
I can use meta add a script with src. But its exec too late due to an additional network requests. 
There is a flicker visible to the naked eye.
```js
useMeta({
  script: [
    {
      src: './rem.js'
    }
  ]
})
```

How can I add code to `<head>` directly?
```html
<head>
<script>
// do something with js
</script>
</head>
```

**Comment by viii47** [comment]
You need to use the `head` or `head()` property. `head()` needs a `return { ... }` embracement which is useful when you need to wait for computed attributes or want to inject dynamic variables at runtime (e.g. item.component which awaits the data of the item)

Example:
```js
head: {
      script: [
        {
          hid: "your-hid",
          src: "yourscript.js",
          async: true|false,
          defer: true|false
        }
      ]
  }

```

**Note:** It seems that hot reload does not work properly sometimes with some new code, so if you see no changes do restart the nuxt client.

**Reply by viii47** [reply]
Edit: Seems this has changed *a lot* -> see https://v3.nuxtjs.org/docs/usage/meta-tags#meta-components

Which would be even easier to include custom third party code

**Reply by NecoHorne** [reply]
Did you manage to figure this out?

In the below example the async script tag with source loads but the next one with inner html script does not load. Still very new to nuxt 3 and the nuxt 2 method does not work, documentation also does not explain how to handle this use case. 

```html
   <Html >

    <Head>

      <Script async src="https://www.googletagmanager.com/gtag/js?id=UA-XXX-X"></Script> <!-- This works -->

      <Script>
        window.dataLayer = window.dataLayer || [];
        function gtag(){dataLayer.push(arguments);}
        gtag('js', new Date());
        gtag('config', 'UA-XXX-X');
      </Script> <!-- this does not work, the tag gets loaded but its empty -->

    </Head>

    </Html>
```

**Reply by viii47** [reply]
Hey @NecoHorne,

in fact I did :) - script content needs to be defined inside a `children` attribute, e.g. `<Script children="....">` 

In your case I'd suggest to define this via `useMeta()` in the setup section. Works of course in the template section too.

### Example

```vue

[...]
<script lang="ts">
export default defineComponent({

setup() {
        useMeta({
            script: [
                {
                    children: 
                    `
                        window.dataLayer = window.dataLayer || [];
                        function gtag(){dataLayer.push(arguments);}
                        gtag('js', new Date());
                        gtag('config', 'UA-XXX-X');
                    `,
                    type: 'text/javascript'
                }

            ]
        })
}

})
</script>

```

**Reply by NecoHorne** [reply]
@viii47 Thank you so much for the help, exactly what I was looking for, still new to vue3 and nuxt3.
